### ISSUE TITLE
Case-sensitiveness of identifiers in spanner-cli statements

### ISSUE THREAD

**Issue by apstndb** [cloudspannerecosystem/spanner-cli#48]
In Cloud Spanner native statements, identifiers are case-insensitive and quoted identifiers are case-sensitive.
It should be emulated in spanner-cli statements.

* Using `INFORMATION_SCHEMA`
  * `SHOW INDEX`
  * `SHOW COLUMNS`
  * ~~`SHOW CREATE TABLES`~~
* Not using `INFORMATION_SCHEMA`
  * `DROP DATABASE`
  * `SHOW CREATE TABLES`

**Comment by yfuruyama** [comment]
Thank you for filing this issue.

As the document mentions, quoted identifiers are not always case sensitive.
https://cloud.google.com/spanner/docs/lexical#identifiers
> Both identifiers and quoted identifiers are case insensitive, with some nuances. See Case Sensitivity for further details.

https://cloud.google.com/spanner/docs/lexical#case_sensitivity
> Table names are usually case insensitive, but may be case sensitive when querying a database that uses case-sensitive table names.

For example, if we have a table called `Albums`, we can run the following query.
```
spanner> select * from `albums`;
+----------+---------+------------+-------------+
| SingerId | AlbumId | AlbumTitle | ReleaseDate |
+----------+---------+------------+-------------+
| 1        | 1       | a1         | 2019-03-08  |
| 1        | 2       | a2         | 2019-03-06  |
| 1        | 3       | a3         | 2019-03-04  |
| 2        | 4       | a4-2       | 2019-03-04  |
+----------+---------+------------+-------------+
4 rows in set (7.57 msecs)
```

So I think we should handle table name as case insensitive for some spanner-cli statements such as `SHOW COLUMNS FROM`, `SHOW INDEXES FROM`, even if it's a quoted table name.

**Comment by yfuruyama** [comment]
But I understand we should fix handling quoted identifier appropriately as current implementation handles the quote (`) as if it's the part of the table name.

e.g.
```
spanner> SHOW COLUMNS FROM `Albums`;
ERROR: table "`Albums`" doesn't exist
```

**Comment by apstndb** [comment]
Thank you for investigating. It seems you are right.
I may have been confused between Cloud Spanner and BigQuery whose table names are [case sensitive](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical?hl=en#case_sensitivity).

refs: https://stackoverflow.com/questions/56478367/when-and-why-are-google-cloud-spanner-table-and-column-names-case-sensitive#comment108066901_56485806

**Comment by apstndb** [comment]
So I will fix https://github.com/cloudspannerecosystem/spanner-cli/pull/49 to tread both quoted identifiers and identifiers as case-insensitive.

**Comment by yfuruyama** [comment]
Sounds good! Thank you for fixing that.

**Comment by apstndb** [comment]
I have finally noticed that table names in SELECT and DMLs are case-insensitive but in DDLs are case-sensitive. It seems more complicated situation.
https://cloud.google.com/spanner/docs/data-definition-language?hl=en#naming_conventions

```
spanner> CREATE TABLE CaseSensitive(pk INT64) PRIMARY KEY(pk);
Query OK, 0 rows affected (7.74 sec)

spanner> DROP TABLE casesensitive;
ERROR: rpc error: code = NotFound desc = Table not found: casesensitive
spanner> DROP TABLE CaseSensitive;
Query OK, 0 rows affected (4.19 sec)
```

I think there are two choices:
* Treat table names as case sensitive
  * Consistent with DDLs
* Treat table names are case insensitive
  * Consistent with SELECT and DMLs
  * Better usability and least surprise(I think)
  * There are no ambiguity because ["No two Cloud Spanner objects can be created with the same name."](https://cloud.google.com/spanner/docs/data-definition-language?hl=en#naming_conventions)

**Comment by apstndb** [comment]
I prefer case-insensitive to case-sensitive.

**Comment by yfuruyama** [comment]
+1 for handling as case insensitive.

Summarizing above, it would be looks like:
| statement | Cloud Spanner | spanner-cli |
|---|---|---|
| SELECT | case insensitive | case insensitive |
| DML (INSERT, UPDATE, DELETE) | case insensitive | case insensitive |
| DDL (CREATE TABLE, etc) | case sensitive | case sensitive |
| SHOW COLUMNS, SHOW INDEXES, etc | N/A | case insensitive |

**Comment by apstndb** [comment]
All statement is fixed in #49
